app/uk/gov/hmrc/customs/notification/services/customsNotificationServices.scala

```diff
@@ -97,7 +97,7 @@ class CustomsNotificationRetryService @Inject()(logger: NotificationLogger,
     val logMsgBuilder = StringBuilder.newBuilder
     notificationWorkItemRepo.saveWithLock(notificationWorkItem).flatMap { workItem =>
       {
```

Comment:
there is a lot of repetition  here. How about something like the below which may allow for more deduplication?
```scala
        if (apiSubscriptionFields.fields.callbackUrl.isEmpty) {
          ("pull", pullClientNotificationRetryService.send(notificationWorkItem))
        } else {
          ("push", pushClientNotificationRetryService.send(apiSubscriptionFields, workItem.item))
        }.flatMap{ tuple =>
          if (tuple._2) {
            //...
          } else {
            if (tuple._1 == "pull") {
               //...
            }
        }
```

Reply:
Optimised